assets/js/components/adminbar/AdminBarApp.js

```diff
@@ -46,15 +47,30 @@ export default function AdminBarApp() {
 	const analyticsModuleConnected = useSelect( ( select ) => select( CORE_MODULES ).isModuleConnected( 'analytics' ) );
 	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
 
+	// Check if each section has zero data.
+	const adminBarImpressionsHasZeroData = useSelect( AdminBarImpressions.hasZeroData );
+	const adminBarClicksHasZeroData = useSelect( AdminBarClicks.hasZeroData );
+	const adminBarUniqueVisitorsHasZeroData = useSelect( AdminBarUniqueVisitors.hasZeroData );
+	const adminBarSessionsHasZeroData = useSelect( AdminBarSessions.hasZeroData );
+
+	// True if _all_ admin bar sections have zero data.
+	const zeroData = ( adminBarImpressionsHasZeroData === true && adminBarClicksHasZeroData === true && adminBarUniqueVisitorsHasZeroData === true && adminBarSessionsHasZeroData === true );
+
 	const onMoreDetailsClick = useCallback( async () => {
 		await trackEvent( 'admin_bar', 'post_details_click' );
 		document.location.assign( detailsURL );
 	}, [ detailsURL ] );
 
+	// Only show the adminbar on valid pages and posts.
 	if ( ! detailsURL || ! currentEntityURL ) {
 		return null;
 	}
 
+	// Don't show the menu until we know there is data.
+	if ( zeroData === undefined ) {
+		return null;
+	}
+
```

Comment:
I added this here because when I was testing each of the sections might load first showing blue warnings briefly before zeroData resolved from undefined to true and the Zero Data component would "pop" in which looked messy. I've removed this and will see if it comes back via QA with suggestions.

Reply:
I think we can enhance this if needed, but that's for another issue I think.